napari/plugins/manager.py

```diff
@@ -0,0 +1,337 @@
+
+        2) Using package metadata:
+            plugins that declare a special key (self.PLUGIN_ENTRYPOINT) in
+            their setup.py `entry_points`.  discovered using `pkg_resources`.
+
+        https://packaging.python.org/guides/creating-and-discovering-plugins/
+
+        Parameters
+        ----------
+        path : str, optional
+            If a string is provided, it is added to sys.path before importing,
+            and removed at the end. by default True
+
+        Returns
+        -------
+        count : int
+            The number of plugin modules successfully loaded.
+        """
+        if path:
+            sys.path.insert(0, path)
+
+        count = 0
+        for plugin_name, module_name in iter_plugin_modules(
+            prefix=self.PLUGIN_PREFIX, group=self.PLUGIN_ENTRYPOINT
+        ):
+            if self.get_plugin(plugin_name) or self.is_blocked(plugin_name):
+                continue
+            try:
+                self._register_module(plugin_name, module_name)
+                count += 1
+            except PluginError as exc:
+                log_plugin_error(exc)
+                self.unregister(name=plugin_name)
+            except Exception as exc:
+                logger.error(
+                    f'Unexpected error loading plugin "{plugin_name}": {exc}'
+                )
+                self.unregister(name=plugin_name)
+
+        if count:
+            msg = f'loaded {count} plugins:\n  '
+            msg += "\n  ".join([n for n, m in self.list_name_plugin()])
+            logger.info(msg)
+
+        if path:
+            sys.path.remove(path)
+
+        return count
+
+    def _register_module(self, plugin_name: str, module_name: str) -> None:
+        """Try to register `module_name` as a plugin named `plugin_name`.
+
+        Parameters
+        ----------
+        plugin_name : str
+            The name given to the plugin in the plugin manager.
+        module_name : str
+            The importable module name
+
+        Raises
+        ------
+        PluginImportError
+            If an error is raised when trying to import `module_name`
+        PluginRegistrationError
+            If an error is raised when trying to register the plugin (such as
+            a PluginValidationError.)
+        """
+        try:
+            mod = importlib.import_module(module_name)
+        except Exception as exc:
+            raise PluginImportError(plugin_name, module_name) from exc
+        try:
+            # prevent double registration (e.g. from entry_points)
+            if self.is_registered(mod):
+                return
+            self.register(mod, name=plugin_name)
+        except Exception as exc:
+            raise PluginRegistrationError(plugin_name, module_name) from exc
+
+
+def entry_points_for(
+    group: str,
+) -> Generator[importlib_metadata.EntryPoint, None, None]:
+    """Yield all entry_points matching "group", from any distribution.
```

Comment:
yeah, I agree the terminology can get a little confusing.  As I understand it, yes "distribution" is basically just an installed package, but here it really refers more to the "dist-info" that accompanies the package.  So I'm trying to use the term distribution to be consistent with how others are... when we're talking more about the metadata in setup.py.
And for example: if a package or module in the environment did _not_ have a `dist-info` folder, or was otherwise lacking that `entry_points.txt` file, it would not show up in this generator.
I added a comment to this effect.